pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -126,36 +127,25 @@ public void initialize(URL location, ResourceBundle resources) {
                            .subscribe(tick -> parent.refreshXPathResults());
 
 
-        xpathExpressionArea.plainTextChanges()
-                           .map(m -> {
-                               if (m.getInsertionEnd() > m.getRemovalEnd()) {
-                                   positionFrom = m.getRemovalEnd();
-                                   positionTo = m.getInsertionEnd();
-                               } else {
-                                   positionFrom = m.getInsertionEnd();
-                                   positionTo = m.getRemovalEnd();
-                               }
-                               if (m.getRemoved().length() > 0) {
-                                   positionFrom--;
-                                   positionTo--;
-                               }
-                               int indexOfSlash = reverseStringSearch('/', xpathExpressionArea.getText(), m.getInsertionEnd());
-                               String input = xpathExpressionArea.getText().substring(
-                                   indexOfSlash + 1, m.getInsertionEnd());
-                               m.getRemovalEnd();
-                               return Tuples.t(indexOfSlash, input);
-                           })
-                           .filter(t -> t._2.matches("[a-zA-Z]+"))
-                           .subscribe(e -> autoComplete(e._1, e._2));
 
-        xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
-            if ((t.isControlDown() && t.getCode().isWhitespaceKey()) || t.getCode() == KeyCode.BACK_SPACE) {
-                autoCompletePopup.show(xpathExpressionArea, xpathExpressionArea.getCharacterBoundsOnScreen(positionFrom, positionTo)
-                                                                               .get()
-                                                                               .getMaxX(), xpathExpressionArea
-                                           .getCharacterBoundsOnScreen(positionFrom, positionTo).get().getMaxY());
-            }
-        });
+        EventStream<Integer> changesEventStream = xpathExpressionArea.plainTextChanges().map(m -> m.getInsertionEnd());
+
+        EventStream<KeyEvent> keyPressed = EventStreams.eventsOf(xpathExpressionArea, KeyEvent.KEY_PRESSED);
+
+        EventStream<Integer> keyCombo = keyPressed
```

Comment:
You don't need to create so many local variables. Just append your treatment here to the previous stream